gmsh_utils/math_utils.py

```diff
@@ -0,0 +1,177 @@
+            eta, xi = point_array[:2]
+        # if the polygon is in the x-z plane, keep the first and third coordinates
+        elif np.isclose(polygon_vertices_array[:, 1], polygon_vertices_array[0, 1]).all():
+            projected_polygon = polygon_vertices_array[:, [0, 2]]
+            eta, xi = point_array[[0, 2]]
+        # if the polygon is in the y-z plane, keep the second and third coordinates
+        elif np.isclose(polygon_vertices_array[:, 0], polygon_vertices_array[0, 0]).all():
+            projected_polygon = polygon_vertices_array[:, [1, 2]]
+            eta, xi = point_array[[1, 2]]
+        # if the polygon is not aligned with any of the planes, rotate the polygon to align it with the x-y plane
+        else:
+            # Calculate the rotation matrix to align the polygon with the x-y plane
+            rotation_matrix = MathUtils.calculate_rotation_matrix_polygon(polygon_vertices_array)
+
+            # Rotate the polygon vertices
+            projected_polygon = np.dot(polygon_vertices_array, rotation_matrix.T)[:,:2]
+
+            # Rotate the point
+            point_array_rot = point_array.dot(rotation_matrix.T)
+
+            # Unpack 2D coordinates
+            eta, xi = point_array_rot[[0, 1]]
+
+        # Shift projected_polygon by one point to the left (circular shift)
+        shifted_polygon = np.roll(projected_polygon, -1, axis=0)
+
+        # Unpack coordinates
+        p1eta, p1xi = projected_polygon[:, 0], projected_polygon[:, 1]
+        p2eta, p2xi = shifted_polygon[:, 0], shifted_polygon[:, 1]
+
+        # Check if the point is within the xi-bounds of the polygon edges
+        between_xi_bounds = ((p1xi < xi + eps) & (xi < p2xi + eps)) | ((p2xi < xi + eps) & (xi < p1xi + eps))
+
+        # Compute the intersection points of the polygon edges with the horizontal line at y
+        eta_intersections = (p2eta - p1eta) * (xi - p1xi) / (p2xi - p1xi) + p1eta
+
+        # check if point is on the edge
+        on_edge = np.isclose(eta, eta_intersections) & between_xi_bounds
+        if on_edge.any():
+            return True
+
+        # Check if the point is to the left of the intersection points
+        left_of_intersection = between_xi_bounds & (eta < eta_intersections+eps)
+
+        # point is inside if an odd number of edges are to the right of the point
+        is_point_inside: bool = np.any(left_of_intersection) & np.sum(left_of_intersection) % 2 == 1
+        return is_point_inside
+
+    @staticmethod
+    def is_point_on_plane(point: npt.NDArray[np.float64],
+                          plane_point: npt.NDArray[np.float64],
+                          plane_normal: npt.NDArray[np.float64]) -> bool:
+        """
+        Check if a point is on a plane.
+
+        Args:
+            - point (npt.NDArray[np.float64]): Point to check.
+            - plane_point (npt.NDArray[np.float64]): Point on the plane.
+            - plane_normal (npt.NDArray[np.float64]): Normal of the plane.
```

Comment:
```suggestion
            - plane_point (npt.NDArray[np.float64]): Points on the plane.
            - plane_normal (npt.NDArray[np.float64]): Vector normal of the plane.
```

Reply:
its meant for a singular point, as the function is written now